### PR TITLE
Feature configure toolchain

### PR DESCRIPTION
**Pull request by dseynhae**
I was unable to completely configure and compile the compiler and libraries on SuSE12.
The main reason is that the Clang compiler driver that we create, keeps falling back on the system installation of GCC (4.8.6), which does not support the C++11 constructs that we need.

This change fixes this by allowing to specify a suitable toolchain from the configuration command line.

Example:

```bash
python3 buildbot/configure.py --gnu-toolchain=/usr/intel/pkgs/gcc/12.1.0
```

This PR also:

* Provides consistent formatting (Black).
* Fixes a typo.

### CODE REVIEW

File: buildbot/configure.py [hunk 1]
```diff
@@ -192,6 +192,10 @@ def do_configure(args):
         )
 
     # Add additional CMake options if provided
+    if args.gnu_toolchain:
+        cmake_cmd.extend(
+            ["-DSYCL_LIBDEVICE_GCC_TOOLCHAIN={}".format(args.gnu_toolchain)]
+        )
```
Comment (bader) [inline buildbot/configure.py:198]: Please, move new code before the comment.

File: buildbot/configure.py [hunk 2]
```diff
@@ -322,6 +326,11 @@ def main():
     parser.add_argument(
         "--shared-libs", action="store_true", help="Build shared libraries"
     )
+    parser.add_argument(
+        "--gnu-toolchain",
+        metavar="GNU_TOOLCHAIN",
+        help="Toolchain for compiler driver result",
```
Comment (bader) [inline buildbot/configure.py:332]: This message might be confusing. This argument passed via `-DSYCL_LIBDEVICE_GCC_TOOLCHAIN`, which (I think) configures GCC toolchain only for `libdevice` project. All other projects will use clang's default configuration. Right?

Comment (dseynhae) [reply buildbot/configure.py:332]: Theoretically, the `SYCL_LIBDEVICE_GCC_TOOLCHAIN` should indeed only apply to the `libdevice`.

However practically, this is the only way I was able get the follow-on `compile.py` to complete. Without this option, the `clang` executables are created correctly. But then those executables fail when deployed for the remaining part of the build: The `clang` executables keep grabbing the system library for `libstdc++` (which fails).

With this setting, we're getting our `clang` compiler to correctly pick up the desired `libstdc++` library...

Comment (bader) [reply buildbot/configure.py:332]: AFAIK, this option doesn't impact the way `clang` executables are created, but how they are used in `libdevice` project. This project uses clang to build SYCL libraries. But help message for `--gnu-toolchain` argument might be interpreted as it changes the location of the GNU toolchain clang picks up by default. I think we should clarify that this option is intended to impact GNU toolchain location for `libdevice` project only.

Comment (dseynhae) [reply buildbot/configure.py:332]: Will do;
I am just getting up to speed on this project, where can I get more details on the different components that we're building here?

Comment (bader) [reply buildbot/configure.py:332]: > where can I get more details on the different components that we're building here?

I don't think we have a separate document answering this question, so the source code for the `configure.py` and CMake files is the best source of information.

I also recommend taking a look at project's documentation - https://intel.github.io/llvm-docs/.

**Comment by dseynhae** [comment]
The problem is better addressed by using macros (CMAKE_C_COMPILER and CMAKE_CXX_COMPILER).

**Pull request closed without merging**